### PR TITLE
Ruby On Rails Budget-app Capstone

### PR DESCRIPTION
**Pull request by hiromataba**
In this PR we did the following :point_down: 
- Here is the [Presentation video](https://www.loom.com/share/7dba8936934144e7bec29423afc0f0d1)
- Interactions
- Splash screen

   - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages

   - The user should be able to register in the app with full name, email and password (all mandatory).
   - The user can log into the app using email and password.
   - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (catagories page)

   - When the user logs in, they are presented with the categories page.
   - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
   - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
   - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page

   - For a given category, the list of transactions is presented, ordered by the most recent.
   - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
   - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page

   - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
   - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
   - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page

   - The user fills out a form to create a new transaction with:
   - name (mandatory)
   - amount (mandatory)
   - categories (mandatory at least one)
   - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
   - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
   - Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: db/migrate/20211116115254_create_join_table_category_deal.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateJoinTableCategoryDeal < ActiveRecord::Migration[6.1]
+  def change
+    create_join_table :categories, :deals do |t|
+      t.index [:category_id, :deal_id]
+      # t.index [:deal_id, :category_id]
```
Comment (Dipeshtwis) [inline db/migrate/20211116115254_create_join_table_category_deal.rb:5]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @hiromataba, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: Your project looks Awesome.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

- [ ] You have done a great job by making the video presentation and also highlighted the feature but one more thing I was expecting please make another video mentioning `a piece of the code you find interesting or is proud of making`.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @hiromataba,

What an awesome project, the UI interface is a beautiful one.
You've done well addressing the concerns of the previous review 👏🏼.
Now consider this project complete. You may merge this branch :shipit:.

Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hiromataba** [comment]
Thanks @Dipeshtwis and @Lameck1 for taking time to review my project :tada: :tada:

**Pull request merged**